gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/service/user/impl/UserServiceImpl.java

```diff
@@ -331,19 +334,22 @@ public Completable forgotPassword(ForgotPasswordParameters params, Client client
                     // narrow users
                     if (users.size() > 1) {
                         // filter by identity provider
-                        if (client.getIdentities() != null && !client.getIdentities().isEmpty()) {
-                            foundUsers = users
-                                    .stream()
-                                    .filter(u -> client.getIdentities().contains(u.getSource()))
-                                    .collect(Collectors.toList());
+                        if (client.getIdentityProviders() != null && !client.getIdentityProviders().isEmpty()) {
+                            foundUsers = users.stream()
+                                    .flatMap(u -> client.getIdentityProviders().stream().map(appIdp -> entry(u, appIdp.getIdentity())))
+                                    .filter(entry -> {
+                                        var user = entry.getKey();
+                                        var identity = entry.getValue();
+                                        return Objects.equals(user.getSource(), identity);
```

Comment:
Maybe statically import this?

Reply:
No possible since `equals` is also a method inherited from `Object`

Reply:
`equals` is also a method from the class `Objects` so we cannot import statically `equals`

Reply:
👍